query/src/main/java/org/apache/kylin/query/util/DefaultQueryTransformer.java

```diff
@@ -50,6 +50,8 @@
     private static final Pattern PIN_SUM_OF_FN_CONVERT = Pattern
             .compile(S0 + "SUM" + S0 + "\\(" + S0 + "\\{\\s*fn" + SM + "convert" + S0 + "\\(" + S0 + "([^\\s,]+)" + S0
                     + "," + S0 + "SQL_DOUBLE" + S0 + "\\)" + S0 + "\\}" + S0 + "\\)", Pattern.CASE_INSENSITIVE);
+    private static final Pattern PTN_HAVING_FUNCTION = Pattern.compile("\\{fn" + "(.*?)" + "\\}",
```

Comment:
It's too general as it matches all functions. How about adding a CURRENT_TIMESTAMP function like PTN_MIN_1 instead?

Reply:
I also need to use TIMESTAMPADD and SQL_TSI_HOUR functions.
Because, Tableau + kylin odbc generate automatically query using this functions.

I don't think that is great idea... I hope that Kylin can support calcite reserved functions. https://calcite.apache.org/docs/reference.html

thank you.